src/sparse_ir/spr.py

```diff
@@ -1,66 +1,135 @@
 # Copyright (C) 2020-2021 Markus Wallerberger and others
 # SPDX-License-Identifier: MIT
 import numpy as np
+from .kernel import LogisticKernel, RegularizedBoseKernel
 from .basis import FiniteTempBasis
-from .sampling import DecomposedMatrix
+from .sampling import MatsubaraSampling, TauSampling, DecomposedMatrix
+from typing import Optional
+
+
+class MatsubaraPoleBasis:
+    def __init__(self, beta: float, poles: np.ndarray) -> None:
+        self._beta = beta
+        self._poles = poles
```

Comment:
I think the `-> None` does not add any information since it clear from the fact that this is an `__init__` method.  Also, I think even with the type annotations, you still need to make sure that the arguments are properly converted

```suggestion
    def __init__(self, beta: float, poles: np.ndarray):
        self._beta = beta
        self._poles = np.array(poles)
```